dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/transform/strategy/BinaryViewStrategy.java

```diff
@@ -0,0 +1,117 @@
+package com.dotmarketing.portlets.contentlet.transform.strategy;
+
+import static java.util.Collections.emptyList;
+import static java.util.Collections.emptyMap;
+
+import com.dotcms.contenttype.model.field.BinaryField;
+import com.dotcms.contenttype.model.field.Field;
+import com.dotcms.util.DotPreconditions;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.DotStateException;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.util.Config;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.UtilMethods;
+import com.liferay.portal.model.User;
+import com.liferay.util.StringPool;
+import io.vavr.control.Try;
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Class intended to collect binaries and present them as Map with entries like `fieldVariable` followed by `Map`
+ */
+public class BinaryViewStrategy extends AbstractTransformStrategy<Contentlet> {
+
+    /**
+     * Regular constructor takes a toolbox
+     * @param toolBox
+     */
+    BinaryViewStrategy(final TransformToolbox toolBox) {
+        super(toolBox);
+    }
+
+    /**
+     * Main Transform function
+     * @param contentlet
+     * @param map
+     * @param options
+     * @param user
+     * @return
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    @Override
+    protected Map<String, Object> transform(final Contentlet contentlet, final Map<String, Object> map,
+            final Set<TransformOptions> options, User user) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'user' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5596402)